fire/fire_test.py

```diff
@@ -288,12 +288,12 @@ def fn3(arg, **kwargs):
                      ('value', {'nothing': False}))
 
   def testTraceFlag(self):
-    self.assertIsInstance(
-        fire.Fire(tc.BoolConverter, 'as-bool True -- --trace'), trace.FireTrace)
-    self.assertIsInstance(
-        fire.Fire(tc.BoolConverter, 'as-bool True -- -t'), trace.FireTrace)
-    self.assertIsInstance(
-        fire.Fire(tc.BoolConverter, '-- --trace'), trace.FireTrace)
+    with self.assertRaisesFireExit(0, 'Fire trace:\n'):
+      fire.Fire(tc.BoolConverter, 'as-bool True -- --trace')
```

Comment:
I know IPython will catch SystemExit exceptions and won't quit. Vanilla `python` quits though. This will also be an issue with `fire.Fire` even without the --trace flag.